internal/contract/types.go

```diff
@@ -127,8 +127,8 @@ func (i *Int32) Get(obj *unstructured.Unstructured) (*int32, error) {
 
 // Set sets the int32 value in the path.
 // Note: Cluster API should never Set values on external objects owner by providers; however this method is useful for writing tests.
-func (i *Int32) Set(obj *unstructured.Unstructured, value int64) error {
-	if err := unstructured.SetNestedField(obj.UnstructuredContent(), value, i.path...); err != nil {
+func (i *Int32) Set(obj *unstructured.Unstructured, value int32) error {
+	if err := unstructured.SetNestedField(obj.UnstructuredContent(), int64(value), i.path...); err != nil {
```

Comment:
is this conversion to int64 correct?

i only ask because of my lack of familiarity with this function, and i had noted all the other conversions from int64 -> int32.

Reply:
this conversion is required because SetNestedField only handles int64 values, so we are up-converting before calling this method